### PR TITLE
capture: some initial cleanups

### PR DESCRIPTION
**Pull request by bluetech**
These are some simple cleanups to capture. I have more complex changes planned, but will send them separately.

The changes are split into independent commits. One of the commits reorders sections which makes the PR diff not very readable.

### CODE REVIEW

**Comment by bluetech** [comment]
There is some problem on Windows. I'll try to figure it out somehow.

File: src/_pytest/capture.py [hunk 1]
```diff
@@ -69,366 +165,357 @@ def pytest_load_initial_conftests(early_config: Config):
...
-        if in_:
-            self.in_ = Capture(0)
-        if out:
-            self.out = Capture(1)
-        if err:
-            self.err = Capture(2)
+    def __init__(self, in_, out, err) -> None:
+        self.in_ = in_
+        self.out = out
+        self.err = err
```
Comment (RonnyPfannschmidt) [inline src/_pytest/capture.py:518]: its great this logic is gone from the ctor, an we move the outside factory functions to a named constructor
to encapsulate the in/out/err  logic

also it looks to me if attrs couldbe used to declare the ctor/non-ctor atributes now

File: src/_pytest/capture.py [hunk 2]
```diff
@@ -497,345 +584,319 @@ def readouterr(self) -> CaptureResult:
...
-            hasattr(self, "_old") and repr(self._old) or "<UNSET>",
-            self._state,
-            self.tmpfile,
-        )
+    # Fixture Control
 
-    def start(self):
-        setattr(sys, self.name, self.tmpfile)
-        self._state = "started"
+    def set_fixture(self, capture_fixture: "CaptureFixture") -> None:
```
Comment (RonnyPfannschmidt) [inline src/_pytest/capture.py:674]: at first glance i believe that pair might yield nicer code if it was a context manager instead

File: testing/test_capture.py [hunk 3]
```diff
@@ -1243,7 +1269,7 @@ def test_capsys_results_accessible_by_attribute(capsys):
 def test_fdcapture_tmpfile_remains_the_same(tmpfile, use):
     if not use:
         tmpfile = True
```
Comment (RonnyPfannschmidt) [inline testing/test_capture.py:1271]: this test looks strange, whats the value of tmpfile?

File: src/_pytest/capture.py (hunk 1 above)
Comment (bluetech) [reply src/_pytest/capture.py:518]: attrs might be nice for some classes here. I'll try to remember this for later.

File: src/_pytest/capture.py (hunk 2 above)
Comment (bluetech) [reply src/_pytest/capture.py:674]: So, it actually was a context manager, but the commits removes it. The reason is that the context manager adds some indirection which makes it more difficult to refactor further (in later PRs).

File: testing/test_capture.py (hunk 3 above)
Comment (bluetech) [reply testing/test_capture.py]: Right, I'm not sure what this was supposed to do either (it's the same in master -- the two parametrization are actually testing the same thing). I added a commit which removes the parametrization.

**Comment by bluetech** [comment]
The Windows failures are caused by commit "capture: fix invalid inheritance between Capture classes". The commit somehow changes the newlines-mode such that on `print("foo")`, instead of capturing `"foo\n"`, `"foo\r\n"` is captured (on Windows).

I've removed the commit now, will resubmit once I have a chance to fix it up.

**Pull request merged**